### PR TITLE
Skip SL build and test if SDK not installed

### LINKED ISSUE #1383: Skip Silverlight build if SDK not installed
In the same way as the CF build is currently handled. 

This was mentioned before, but I don't think an issue was made. It's just bit me again, so thought I'd fix it, if that's still considered the right approach?

### PR DESCRIPTION
**Pull request by ChrisMaddock**
Fixes #1383.

The changes are all in the first commit, the second commit it just converting the mix of tabs/spaces to all spaces - can strip that one off if it's not wanted! The code is basically a carbon copy of what's in place for CF.

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -73,13 +73,10 @@ important top-level tasks to use are listed here:
...
- 2. If the compact framework is not installed on a machine, building and testing tasks
-    for CF are skipped and a warning is issued.
+ 2. If the compact framework or Silverlight SDK are not installed on a machine, the relevant 
+    building and testing tasks are skipped and a warning is issued.
 
- 3. Currently, Silverlight must be installed or an error occurs when trying to build or
-    test it. We will be changing this soon to match the strategy for CF.
-
- 4. The Test and TestAll targets currently due the same thing. Both are retained for backward
+ 3. The Test and TestAll targets currently due the same thing. Both are retained for backward
```
Comment (CharliePoole) [inline BUILDING.md]: s/due/do/
remove one "backward"

Comment (ChrisMaddock) [inline BUILDING.md]: Good spot - just changed the numbers on these lines while skimming over! Will sort these later in a new pull

Comment (ChrisMaddock) [inline BUILDING.md]: #1402

**Comment by CharliePoole** [comment]
@ChrisMaddock We frequently would edit a small typo like this online, right in the PR. If using the nunit org for your branch, any of us could do that. I don't have experience using a fork, but I think at least you could do the same. It's a way to avoid an extra push or PR for a trivial change.

**Comment by rprouse** [comment]
@ChrisMaddock you are welcome to continue working in your forks if you prefer, but you have write access to all the NUnit repositories. You cannot push to master without a PR so it is safe and it allows other team members to contribute to your branches if needed. It also makes it easier to keep your source in sync since you don't have to pull from origin and push to your fork.

As I said, it is up to you, I just wanted to let you know it is okay to push branches directly to the main repository.

**Comment by ChrisMaddock** [comment]
@CharliePoole - great, wasn't aware I could do that after it was merged. To do that, would I just add an extra commit to the branch, and then remerge?

@rprouse - I've kept away as I'm a git novice - always used mercurial professionally! It does seem like it would make things easier however, so I'll start doing so. :-)

**Comment by rprouse** [comment]
@ChrisMaddock you needn't worry too much. As I said, we have the master branch locked, so all you can screw up is your own branches. We've all done that a few times :smile:

**Comment by ChrisMaddock** [comment]
Wasn't aware of that - that makes things somewhat less daunting!

**Comment by CharliePoole** [comment]
@ChrisMaddock Not after you merge. But while you have a PR, you can look at the files tab and just click the edit button on the file you want to change. Obviously, it should only be done for small changes. When reviewing a PR for an nunit branch to merge into master, I will sometimes do that at the same time I make a comment about a misspelling.

Alternatively, you just create a new commit and push it. The PR gets updated.

**Comment by ChrisMaddock** [comment]
Great - thanks, will do that next time.

**Pull request merged**